artichoke-backend/src/extn/core/time/args.rs

```diff
@@ -57,18 +58,64 @@ impl TryConvertMut<&mut [Value], Args> for Artichoke {
-                    let arg = to_int(self, arg)?;
-                    let arg: i64 = arg.try_convert_into(self)?;
-
-                    result.month = match u8::try_from(arg) {
+                    // ```irb
+                    // 3.1.2 => Time.utc(2022, 2).month
+                    // => 2
+                    // 3.1.2 => class I; def to_int; 2; end; end
+                    // => :to_int
+                    // 3.1.2 => Time.utc(2022, I.new).month
+                    // => 2
+                    // 3.1.2 > Time.utc(2022, "feb").month
+                    // => 2
+                    // 3.1.2 > class A; def to_str; "feb"; end; end
+                    // => :to_str
+                    // 3.1.2 > Time.utc(2022, A.new).month
+                    // => 2
+                    // 3.1.2 > class I; def to_str; "2"; end; end
+                    // => :to_str
+                    // 3.1.2 > Time.utc(2022, I.new).month
+                    // => 2
+                    // ```
+                    let month: i64 = if Ruby::Fixnum == arg.ruby_type() {
+                        // Short circuit to avoid string checking
+                        let arg = to_int(self, arg)?;
+                        arg.try_convert_into(self)?
+                    } else {
+                        if let Ok(arg) = to_str(self, arg) {
+                            let mut month_str: Vec<u8> = arg.try_convert_into_mut(self)?;
+                            month_str.make_ascii_lowercase();
+                            match month_str.as_slice() {
+                                b"jan" => Ok(1),
+                                b"feb" => Ok(2),
+                                b"mar" => Ok(3),
+                                b"apr" => Ok(4),
+                                b"may" => Ok(5),
+                                b"jun" => Ok(6),
+                                b"jul" => Ok(7),
+                                b"aug" => Ok(8),
+                                b"sep" => Ok(9),
+                                b"oct" => Ok(10),
+                                b"nov" => Ok(11),
+                                b"dec" => Ok(12),
+                                _ => {
+                                    // Delegate to `Kernel#Integer` as last
+                                    // resort to handle Integer strings.
+                                    let arg = integer(self, arg, None)?;
+                                    arg.try_convert_into(self)
+                                }
+                            }
+                        } else {
+                            let arg = to_int(self, arg)?;
+                            arg.try_convert_into(self)
+                        }?
+                    };
+
+                    result.month = match u8::try_from(month) {
                         Ok(month @ 1..=12) => Ok(month),
                         _ => Err(ArgumentError::with_message("mon out of range")),
```

Comment:
similarly, this and other `match`es in this file would be structured:

```suggestion
                        Ok(month @ 1..=12) => month,
                        _ => return Err(ArgumentError::with_message("mon out of range").into()),
```